resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventInputImpl.java

```diff
@@ -82,139 +90,51 @@ public InboundSseEvent read() throws IOException
+         if (line.isEmpty() && fieldSet)
          {
-            if (currentState == SseConstants.EVENT.START)
-            {
-               if (b == '\r' || b == '\n')
-               {
-                  continue;
-               }
-
-               if (b == ':')
-               {
-                  currentState = SseConstants.EVENT.COMMENT;
-                  continue;
-               }
-               else
-               {
-                  temSave.write(b);
-                  currentState = SseConstants.EVENT.FIELD;
-                  continue;
-               }
-            }
-            if (currentState == SseConstants.EVENT.COMMENT)
+            InboundSseEventImpl event = (InboundSseEventImpl) eventBuilder.build();
+            if (this.providers != null)
             {
-
-               b = readLine(entityStream, '\n', temSave);
-               String commentLine = temSave.toString(charset.name());
-               eventBuilder.commentLine(commentLine);
-               temSave.reset();
-               currentState = SseConstants.EVENT.START;
-               continue;
-            }
-            if (currentState == SseConstants.EVENT.FIELD)
-            {
-               temSave.write(b);
-               b = readLine(entityStream, ':', temSave);
-               String fieldName = temSave.toString(StandardCharsets.UTF_8.name());
-               temSave.reset();
-               if (b == ':')
-               {
-                  //spec says there is space after colon
-                  do
-                  {
-                     b = entityStream.read();
-                  }
-                  while (b == ' ');
-
-                  if (b != '\n' && b != '\r' && b != -1)
-                  {
-                     temSave.write(b);
-                     b = readLine(entityStream, '\n', temSave);
-                  }
-               }
-
-               processField(eventBuilder, fieldName, mediaType, temSave.toByteArray());
-               temSave.reset();
-               currentState = SseConstants.EVENT.START;
-               continue;
+               event.setProvider(this.providers);
             }
+            return event;
          }
-         catch (IOException e)
+         if (!line.isEmpty())
          {
-            throw new IOException(Messages.MESSAGES.readEventException(), e);
-         }
-      }
-      InboundSseEventImpl event = (InboundSseEventImpl) eventBuilder.build();
-      if (this.providers != null)
-      {
-         event.setProvider(this.providers);
-      }
-      return event;
-   }
-
-   private int readLine(final InputStream in, final int delimiter, final OutputStream out) throws IOException
-   {
-      int b;
-      while ((b = in.read()) != -1)
-      {
-         if (!textLike && escape && b == '\\')
-         {
-            b = in.read();
-            if (b != '\\' && b != '\n' && b != '\r')
+            int colon = line.indexOf(":");
+            String filedValue = line.substring(colon + 2);
```

Comment:
This probably breaks if a line contains a colon as the last char. You should check on the colon index before doing a substring.

Reply:
Thanks for the review. I'll fix this.